src/lib/stream.ts

```diff
@@ -0,0 +1,66 @@
+import { pipeline } from 'node:stream/promises'
+
+import type { Handler, HandlerEvent, HandlerContext, StreamingHandler, StreamingResponse } from '../function/index.js'
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace awslambda {
+    function streamifyResponse(
+      handler: (event: HandlerEvent, responseStream: NodeJS.WritableStream, context: HandlerContext) => Promise<void>,
+    ): Handler
+
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace HttpResponseStream {
+      function from(stream: NodeJS.WritableStream, metadata: Omit<StreamingResponse, 'body'>): NodeJS.WritableStream
+    }
+  }
+}
+
+/**
+ * Enables streaming responses. `body` now accepts a `Readable` stream or Web stream.
+ *
+ * @example
+ * ```
+ * const { Readable } = require('stream');
+ *
+ * export const handler = stream(async (event, context) => {
+ *   const stream = Readable.from(Buffer.from(JSON.stringify(event)))
+ *   return {
+ *     statusCode: 200,
+ *     body: stream,
+ *   }
+ * })
+ * ```
+ *
+ * @example
+ * ```
+ * export const handler = stream(async (event, context) => {
+ *   const response = await fetch('https://api.openai.com/', { ... })
+ *   // ...
+ *   return {
+ *     statusCode: 200,
+ *     body: response.body, // Web stream
```

Comment:
Not sure I understand the differentiation between a `ReadableStream` and a Web stream, since `response.body` is a `ReadableStream`?

Reply:
`Readable` is the old NodeJS class, and `ReadableStream` is the WHATWG interface that's been added in Node v18 under `node:stream/web`. Node.JS refers to the streams from WHATWG Fetch spec as `Web Streams`.

Reply:
Ah, I hadn't realised that was the term they used. I left a suggestion for using WHATWG instead which I think might be slightly clearer, but feel free to take it or leave it.